operate/webapp/src/main/java/io/camunda/operate/webapp/zeebe/operation/adapter/ServicesBasedAdapter.java

```diff
@@ -0,0 +1,380 @@
+    final var variableDocumentRecord =
+        // operationId is not used in the updateJob service method
+        executeCamundaServiceAnonymously(
+            (authentication) ->
+                withOperationReference(
+                    operationReference ->
+                        elementInstanceServices.setVariables(
+                            new SetVariablesRequest(
+                                scopeKey, variables, local, operationReference)),
+                    operationId));
+    return variableDocumentRecord.getScopeKey();
+  }
+
+  protected static <T> CompletableFuture<T> withOperationReference(
+      final Function<Long, CompletableFuture<T>> command, final String id) {
+    try {
+      final long operationReference = Long.parseLong(id);
+      return command.apply(operationReference);
+    } catch (final NumberFormatException e) {
+      LOGGER.debug(
+          "The operation reference provided is not a number: {}. Ignoring propagating it to zeebe commands.",
+          id);
+    }
+    return null;
+  }
+
+  private <T> T executeCamundaServiceAnonymously(
+      final Function<Authentication, CompletableFuture<T>> method) {
+    return executeCamundaService(method, RequestMapper.getAnonymousAuthentication());
+  }
+
+  private <T> T executeCamundaService(
+      final Function<Authentication, CompletableFuture<T>> method,
+      final Authentication authentication) {
+    try {
+      return method.apply(authentication).join();
+    } catch (final Exception e) {
+      throw handleException(e);
+    }
+  }
+
+  private RuntimeException handleException(final Throwable error) {
+    return switch (error) {
+      case final CompletionException ce -> handleException(ce.getCause());
+      case final CamundaBrokerException cbe -> mapCamundaBrokerException(cbe);
+      default -> new RuntimeException("Failed to execute request: " + error.getMessage(), error);
+    };
+  }
+
+  private RuntimeException mapCamundaBrokerException(final CamundaBrokerException exception) {
+    if (exception.getCause() instanceof final BrokerRejectionException brokerRejection) {
+      final var rejection = brokerRejection.getRejection();
+      final String message =
+          String.format(
+              "Request '%s' rejected with code '%s': %s",
+              rejection.intent(), rejection.type(), rejection.reason());
+      final var type = rejection.type();
+      if (type.equals(RejectionType.NOT_FOUND)) {
+        return new NotFoundException(message, exception);
+      }
+      if (type.equals(RejectionType.UNAUTHORIZED) || type.equals(RejectionType.FORBIDDEN)) {
+        return new NotAuthorizedException(message, exception);
+      }
+    }
+    if (exception.getCause() instanceof final BrokerErrorException brokerError) {
+      final var errorCode = brokerError.getError().getCode();
+      if (errorCode.equals(ErrorCode.PROCESS_NOT_FOUND)) {
+        return new NotAuthorizedException("Process not found", exception);
```

Comment:
❓ Should it be a `NotFoundException`?

Reply:
I removed this code and added exception evaluation based on the [RestErrorMapper](https://github.com/camunda/camunda/blob/3bd10722dc2a4ec83c304df6e9daa89deeab55e1/zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RestErrorMapper.java#L87).